ui/pages/settings/contact-list-tab/add-contact/add-contact.component.js

```diff
@@ -66,22 +67,29 @@ export default class AddContact extends PureComponent {
     }
   };
 
+  onChange = (input) => {
+    this.setState({ input });
+    this.dValidate(input);
+  };
+
   renderInput() {
     return (
       <EnsInput
         scanQrCode={(_) => {
           this.props.scanQrCode();
         }}
-        onChange={this.dValidate}
+        onChange={this.onChange}
         onPaste={(text) => this.setState({ ethAddress: text })}
-        onReset={() => this.setState({ ethAddress: '', ensAddress: '' })}
+        onReset={() =>
+          this.setState({ ethAddress: '', ensAddress: '', input: '' })
+        }
         updateEnsResolution={(address) => {
           this.setState({ ensAddress: address, error: '', ensError: '' });
         }}
         updateEnsResolutionError={(message) =>
           this.setState({ ensError: message })
         }
-        value={this.state.ethAddress || ''}
+        userInput={this.state.ethAddress || this.state.input}
```

Comment:
So previously, changing the `ethAddress` state would also update the `value` prop and trigger `onChange` to be called, which would validate the new address. Now that validation is being skipped onPaste, and when a new QR Code is received, because that weird cycle isn't happening anymore.

Reply:
This lead to me discovering completely broken contact list/ add new contact. This is resolved now and I addressed this feedback (i think)

Reply:
https://github.com/MetaMask/metamask-extension/pull/10965/commits/49c4a0f03da64b116efbe7ecdf249c1c84ea7e0d